chainerx_cc/chainerx/cuda/cuda_device/linalg.cu

```diff
@@ -204,5 +284,142 @@ public:
 
 CHAINERX_CUDA_REGISTER_KERNEL(InverseKernel, CudaInverseKernel);
 
+class CudaSVDKernel : public SVDKernel {
+public:
+    std::tuple<Array, Array, Array> Call(const Array& a, bool full_matrices, bool compute_uv) override {
+        Device& device = a.device();
+        Dtype dtype = a.dtype();
+        CudaSetDeviceScope scope{device.index()};
+
+        CHAINERX_ASSERT(a.ndim() == 2);
+
+        int64_t n = a.shape()[0];
+        int64_t m = a.shape()[1];
+
+        Array x = Empty(Shape{n, m}, dtype, device);
+        bool trans_flag;
+
+        if (m >= n) {
+            device.backend().CallKernel<CopyKernel>(a, x);
+            trans_flag = false;
+        } else {
+            m = a.shape()[0];
+            n = a.shape()[1];
+            x = x.Reshape(Shape{n, m});
```

Comment:
Is this reshape correct?

Reply:
Yes... When `a.shape()[1] >= a.shape()[0]` array `x == a` and has the shape of `a.shape()`, if `a.shape()[1] < a.shape()[0]` then `x == a.Transpose()`